pytest_plugins/rerun_rp/rerun_rp.py

```diff
@@ -29,24 +29,47 @@ def _get_failed_tests(launch, fail_args):
     return tests
 
 
-def _get_test_collection(rp_failed_tests, items):
+def _get_skipped_tests(launch):
+    """Returns skipped test names from Report Portal Launch based on arguments"""
+    tests = launch.tests(status='skipped').keys()
+    # Formating test names for 'member of pytest test items' operation
+    tests = [str(test).replace('::', '.') for test in tests]
+    return tests
+
+
+def _get_test_collection(selectable_tests, items):
     """Returns the selected and deselected items"""
     # Select test item if its in failed tests else deselect
     LOGGER.debug('Selecting/Deselecting tests based on latest launch test results..')
     selected = []
     deselected = []
     for item in items:
         test_item = f"{item.location[0]}.{item.location[2]}"
-        if test_item in rp_failed_tests:
+        if test_item in selectable_tests:
             selected.append(item)
         else:
             deselected.append(item)
     return selected, deselected
 
 
+def _validate_launch(launch, sat_version):
+    """Stop session based on RP Launch statistics and info"""
+    if launch.info['isProcessing']:
+        raise LaunchError(f'The launch of satellite version {sat_version} is not Finished yet')
+    fail_percent = round(
+        (int(launch.statistics['failed']) / int(launch.statistics['total']) * 100)
+    )
+    if fail_percent > 20:
+        raise LaunchError(
+            f'The latest launch of Satellite verson {sat_version} has {fail_percent}% tests '
+            'failed. Which is higher than the threshold of 20%. '
+            'Examine the failures thoroughly and check for any major issue.'
+        )
+
+
 def pytest_addoption(parser):
     """Add options for pytest to collect only failed tests"""
-    help_script = f'''
+    fhelp_script = f'''
```

Comment:
Can't you leave that as `help_script` and just assign new value to variable after first `parser.addoption`?

`fhelp_script` and `shelp_script` don't strike me as particularly readable names.